File: use-cases/school-timetabling/src/main/java/org/acme/schooltimetabling/rest/exception/TimetableSolverException.java
package org.acme.schooltimetabling.rest.exception;

import jakarta.ws.rs.core.Response;

public class TimetableSolverException extends RuntimeException {

    private final String jobId;

    private final Response.Status status;

    public TimetableSolverException(String jobId, Response.Status status, String message) {
        super(message);
        this.jobId = jobId;
        this.status = status;
    }

    public TimetableSolverException(String jobId, Throwable cause) {
        super(cause.getMessage(), cause);
        this.jobId = jobId;
        this.status = Response.Status.INTERNAL_SERVER_ERROR;
    }

    public String getJobId() {
        return jobId;
    }

    public Response.Status getStatus() {
        return status;
    }
}

File: quickstarts/school-timetabling-kotlin/src/main/kotlin/org/acme/kotlin/schooltimetabling/rest/exception/TimetableSolverException.kt
package org.acme.kotlin.schooltimetabling.rest.exception

import jakarta.ws.rs.core.Response

class TimetableSolverException : RuntimeException {

    var jobId: String

    var status: Response.Status

    constructor(jobId: String, status: Response.Status, message: String?) : super(message) {
        this.jobId = jobId
        this.status = status
    }

    constructor(jobId: String, cause: Throwable) : super(cause.message, cause) {
        this.jobId = jobId
        this.status = Response.Status.INTERNAL_SERVER_ERROR
    }
}
